### ISSUE TITLE
'bad adjusted gamma' occur

### ISSUE THREAD

**Issue by astratsi** [pfmonville/whole_history_rating#10]
The data exceeds 3200 games, and during iteration, a 'bad adjusted gamma' error occurs. 
After the error appears, the results show that a specific player has an abnormally high rating.

{
    "499": **15820.314153319185**,
    "300": 624.9808229062172,
    "299": 591.5682009170249,
    "36": 578.8136388010431,
    "655": 577.0898912577412,
    "347": 541.4121475900309,

The error occurs regardless of whether the iteration is done automatically, manually, or manually with 10 iterations.
There are no issues during the game input process, but the error occurs during the iteration.

[whr.zip](https://github.com/user-attachments/files/16488011/whr.zip)

Here is the data insert code and the game result files. The data insert is designed to load and process data by folders, and it works well for 1991 and 1992. However, in 1993, a 'bad adjusted gamma' error occurs. The error has been identified to occur in the '19931105_WOC_1993.ELO' file, but running this file individually or processing only the 1993 data works fine, making it difficult to pinpoint the exact cause.

Inputting the 1991 folder => Successfully completed
Inputting the 1992 folder (loading 1991.whr) => Successfully completed
Inputting the 1993 folder (loading 1992.whr) => 'bad adjusted gamma' error occurs
Inputting the 1993 folder without loading an existing whr => Successfully completed

**Comment by pfmonville** [comment]
Hi,

I tried running your script with the provided data as is. However, I encountered recursion limit errors with pickle (which I plan to investigate further).

I processed all the files at once (covering the years 1991, 1992, and 1993) and did not encounter the ‘bad adjusted gamma’ error. Here is the resulting JSON file (excluding names):
[results-1991-1992-1993.json](https://github.com/user-attachments/files/16488427/results-1991-1992-1993.json)

I ran this on a Silicon MacBook Pro and the version 1.6.2.
